### PR TITLE
set up CI to work w/pre-release OpenStudio packages by adding OPENSTUDIO_VERSION_EXT.

### PR DESCRIPTION
**Pull request by anyaelena**
NOTE: adding separate issue/PR to allow Dockerfile to build off of pre-release OpenStudio Docker images.   Changes to Dockerfile are not appropriate at this stage in the 2.8.0 release.

### CODE REVIEW

**Comment by anyaelena** [comment]
we are going to wait until after 2.8.0 to tackle pre-release extensions.  for now, tim will also publish a duplicate package in the legacy format (ie $VERSION.$SHA instead of $VERSION$EXT.$SHA)

**Comment by nllong** [comment]
@anyaelena - can you resolve these conflicts and I will review this.

**Comment by anyaelena** [comment]
ok @nllong this is also ready for review

publishing of OSS docker images and amis w/version extension should just work from master branch, assuming VERSION_EXT is set in server/lib/version.rb.

**Comment by anyaelena** [comment]
Fixes #474

**Comment by anyaelena** [comment]
actually @nllong hold off a little longer.  i'm looking at moving our version/sha/ext config into a single yml file that can be referenced in travis, appveyor, and Dockerfile

**Comment by anyaelena** [comment]
@nllong  please review. Fixes #474.

File: .travis.yml [hunk 1]
```diff
@@ -56,13 +56,14 @@ jobs:
         - BUILD_TYPE=docker
       before_script:
         # remove the .git in the .dockerignore so coverage can be reported correctly to coveralls
+        - export OPENSTUDIO_TAG=$(ruby -e "load 'server/app/lib/openstudio_server/version.rb'; print OpenstudioServer::VERSION+OpenstudioServer::VERSION_EXT")
```
Comment (nllong) [inline .travis.yml]: nice

File: Dockerfile [hunk 2]
```diff
@@ -3,8 +3,8 @@
 # TO_BUILD_AND_RUN: docker-compose up
 # NOTES:            Currently this is one big dockerfile and non-optimal.
 
-
-ARG OPENSTUDIO_VERSION=2.8.0
+#may include suffix
+ARG OPENSTUDIO_VERSION
```
Comment (nllong) [inline Dockerfile]: looks like this is now going to be a required input... that is if the user does not specify the input then this won't build.

File: server/app/lib/openstudio_server/version.rb [hunk 3]
```diff
@@ -36,6 +36,7 @@
 module OpenstudioServer
   VERSION = '2.8.0'.freeze
   # format should be ^.*\-{1}[a-z]+[0-9]+
-  # for example: -rc0, -beta6, -customusecase0
+  # for example: -rc1, -beta6, -customusecase0
   VERSION_EXT = ''.freeze # with preceding - or +
+  OS_SHA = '55665635f0'
```
Comment (nllong) [inline server/app/lib/openstudio_server/version.rb]: this will need to be specified every version change?

**Comment by anyaelena** [comment]
closes #474

**Pull request merged**